functions/Set-DbaCompression.ps1

```diff
@@ -0,0 +1,508 @@
+	
+	.PARAMETER Database
+		The database(s) to process - this list is autopopulated from the server. If unspecified, all databases will be processed.
+	
+	.PARAMETER ExcludeDatabase
+		The database(s) to exclude - this list is autopopulated from the server
+	
+	.PARAMETER IncludeSystemDBs
+		Switch parameter that when used will display system database information
+	
+	.PARAMETER Silent
+		Replaces user friendly yellow warnings with bloody red exceptions of doom!
+		Use this if you want the function to throw terminating errors you want to catch.
+    
+    .PARAMETER MaxRunTime
+		Will continue to Alter tables and indexes for the give amount of minutes.
+
+    .PARAMETER PercentCompression
+		Will only work on the tables/indexes that have the calulated savings at and higer for the given number provided.
+	
+	.NOTES
+		Author: Jason Squires (@js_0505, jstexasdba@gmail.com)
+		Tags: Compression, Table, Database
+		Website: https://dbatools.io
+		Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+		License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+	
+	.LINK
+		https://dbatools.io/Set-DbaCompression
+	
+	.EXAMPLE
+		Set-DbaCompression -SqlInstance localhost -MaxRunTime 60 -PercentCompression 25
+		Set the compression run time to 60 minutes and will start the compression of of tables/indexes
+        that have a difference of 25% or higher between current and recommended.
+	
+	.EXAMPLE
+		Set-DbaCompression -SqlInstance ServerA -Database DBName -MaxRunTime 60 -PercentCompression 25 | Out-GridView
+		Set the compression run time to 60 minutes and will start the compression of of tables/indexes
+        that have a difference of 25% or higher between current and recommended and the results into and nicely formated GridView.
+	
+	.EXAMPLE
+		Set-DbaCompression -SqlInstance ServerA -MaxRunTime 60 -PercentCompression 25
+		Set the compression run time to 60 minutes and will start the compression of of tables/indexes; across all databases;
+        that have a difference of 25% or higher between current and recommended.
+	
+    .EXAMPLE
+        $servers = 'Server1','Server2'
+        foreach ($svr in $servers)
+        {
+			Set-DbaCompression -SqlInstance $svr -MaxRunTime 60 -PercentCompression 25 | Export-Csv -Path C:\temp\CompressionAnalysisPAC.csv -Append
+        }
+	
+	    This produces a full list of all your servers listed and is pushed to a csv for you to analyize.
+        Set the compression run time to 60 minutes and will start the compression of of tables/indexes; across all listed servers;
+        that have a difference of 25% or higher between current and recommended.
+
+#>
+	[CmdletBinding(DefaultParameterSetName = "Default")]
+	param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]
+		$SqlInstance,
+		
+		[System.Management.Automation.PSCredential]
```

Comment:
We are setting this to just `[PSCredential]` to prevent issues on Windows 7 ( think it was).